services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/table/TableViewManagerImpl.java

```
@@ -633,53 +616,104 @@ BucketAndKey createViewSnapshotAndUploadToS3(IdAndVersion idAndVersion, ViewScop
 					writer.writeNext(nextLine);
 				};
 				// write the snapshot to the temp file.
-				indexManager.createViewSnapshot(idAndVersion.getId(), scopeType, viewSchema,
-						writerAdapter);
+				schema = indexManager.streamTableToCSV(idAndVersion, writerAdapter);
 			}
 			// upload the resulting CSV to S3.
-			String key = idAndVersion.getId() + "/" + UUID.randomUUID().toString() + ".csv.gzip";
-			String bucket = config.getViewSnapshotBucketName();
 			s3Client.putObject(new PutObjectRequest(bucket, key, tempFile));
-			return new BucketAndKey().withBucket(bucket).withtKey(key);
-		} catch (UndefinedViewScopeException e) {
-			// PLFM-7417 - Contextually update the error message based on the entity type.
-			EntityType entityType = nodeManager.getNodeType(KeyFactory.keyToString(idAndVersion.getId()));
-			if (EntityType.dataset.equals(entityType)) {
-				throw new UndefinedViewScopeException("You cannot create a version of an empty Dataset. Add files to this Dataset before creating a version.");
-			} else if (EntityType.datasetcollection.equals(entityType)) {
-				throw new UndefinedViewScopeException("You cannot create a version of an empty Dataset Collection. Add Datasets to this Dataset Collection before creating a version.");
-			}
-			throw e;
- 		} catch (IOException e) {
+		} catch (IOException e) {
 			throw new RuntimeException(e);
 		} finally {
 			// unconditionally delete the temporary file.
 			if (tempFile != null) {
 				tempFile.delete();
 			}
 		}
+		return schema;
+	}
+	
+	void validateViewForSnapshot(IdAndVersion idAndVersion) throws TableUnavailableException {
+		ValidateArgument.required(idAndVersion, "The view id");
+		
+		// This will make sure to trigger building the view if not available
+		TableStatus status = tableManagerSupport.getTableStatusOrCreateIfNotExists(idAndVersion);

```

If we use tableManagerSupport.getTableStatusState() that does not trigger an update, we can just throw an illegalarg if not available and not worry about the act of snapshoting trigger any type of change to the view.